### DISCUSSION TITLE
How to end conversation and transfer control to another handler

### DISCUSSION THREAD

**Discussion by avfawkes** [python-telegram-bot/python-telegram-bot#3227] [Q&A]
Hi, there
When user inside ConversationHandler and enter some command which outside, its obviously not handled.  For example after "/bug" command user text "/start" and "unknown_command" triggered. How to forward it to start handler?
P.S. groups not suitable because of logic, its just piece of code.

```
bug_conv_handler = ConversationHandler(
        entry_points=[CommandHandler("bug", bug_report_entry_point)],
        states={FEEDBACK: [MessageHandler(~filters.COMMAND, bug_report_feedback)]},
        fallbacks=[CommandHandler("cancel", conversation_cancel),
                   CommandHandler("help", bug_help),
                   MessageHandler(filters.COMMAND, unknown_command)], allow_reentry=True)
application.add_handler(bug_conv_handler)
application.add_handler(CommandHandler("start", start))
```

**Comment by Poolitzer** [comment]
If you don't want the CH to handle it, dont add it to the fallbacks?

**Reply by Bibo-Joshi** [reply]
or putbut the `/start` handler in a different group. Please have a look at the discussion in #1640 , the docs of `add_handler` and the wiki page on "Frequently requested design patterns", which has a section on "handling updates in several handlers".

**Comment by avfawkes** [comment]
Put handlers in different groups do not solve task of closing conversation, when top level command typed.

Found a solution to add "return ConversationHandler.END" to every top level commands and put it to fallback. If command typed without conversation it rise no error, if user inside conversation type top level command, it handled and close conversation. Is such pattern Ok?

```
bug_conv_handler = ConversationHandler(
        entry_points=[CommandHandler("bug", bug_report_entry_point)],
        states={FEEDBACK: [MessageHandler(~filters.COMMAND, bug_report_feedback)]},
        fallbacks=[CommandHandler("cancel", conversation_cancel),
                   CommandHandler("help", bug_help),
                   CommandHandler("bd", bday),
                   MessageHandler(filters.COMMAND, unknown_command)], allow_reentry=True)
application.add_handler(bug_conv_handler)
application.add_handler(CommandHandler("bd", bday))

async def bday(update: Update, context: ContextTypes.DEFAULT_TYPE) -> int:
    pass
    return ConversationHandler.END
```
P.S. By the way, is it Ok to create handlers each time it needed? I mean "CommandHandler("bd", bday)" here in 2 places? or better to save to variable?

**Reply by Lodimup** [reply]
I'd like to add another note.

It must be  registered after the Convention handler
```
    watch_conv_handler = ConversationHandler(
   ...
        fallbacks=[
            CommandHandler("cancel", cancel),
            CommandHandler("start", start),
        ],
    )
    app.add_handler(watch_conv_handler)

 app.add_handler(CommandHandler("start", start))
```

**Reply by ergobound** [reply]
Thank you! That worked for me.
I put a little lower this line
`app.add_handler(CommandHandler('start', start))`
